frontend/src/__tests__/cypress/cypress/pages/projects.ts

```diff
@@ -183,6 +221,49 @@ class ProjectDetailsSettingsTab extends ProjectDetails {
   }
 }
 
+class DataConnectionModal extends Modal {
```

Comment:
I feel like this can live in its own file for `dataConnections` rather than in the projects file. This is getting large and would help with discovery.